packages/experimental/Menu/src/MenuTrigger/MenuTrigger.tsx

```diff
@@ -17,11 +18,33 @@ export const MenuTrigger = stagedComponent((props: MenuTriggerProps) => {
     }
 
     const child = childrenArray[0];
-    const revised = React.cloneElement(child, menuTrigger.props);
+    const revisedState = getRevisedState(menuTrigger, child.props);
+    const revised = React.cloneElement(child, revisedState);
 
     return <MenuTriggerProvider value={menuTrigger.hasSubmenu}>{revised}</MenuTriggerProvider>;
   };
 });
 MenuTrigger.displayName = menuTriggerName;
 
+const getRevisedState = memoize(getRevisedStateWorker);
+function getRevisedStateWorker(state: MenuTriggerState, props: any): MenuTriggerState {
+  const revisedState = { ...state };
```

Comment:
Could you leave a comment explaining this is needed because children isn't available in outer render? That wasn't clear to me reading this.